File: pubsub/src/main/java/glitch/pubsub/events/UnknownResponseEvent.java
package glitch.pubsub.events;

import com.google.gson.JsonElement;
import glitch.api.ws.events.AbstractEvent;
import glitch.api.ws.events.IEvent;
import glitch.pubsub.GlitchPubSub;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class UnknownResponseEvent extends AbstractEvent<GlitchPubSub> implements IEvent<GlitchPubSub> {
    private final JsonElement data;
    public UnknownResponseEvent(GlitchPubSub client, JsonElement data) {
        super(client);
        this.data = data;
    }
}

File: pubsub/src/main/kotlin/glitch/pubsub/events/UnknownResponseEvent.kt
package glitch.pubsub.events

import com.google.gson.JsonElement
import glitch.api.ws.events.AbstractEvent
import glitch.api.ws.events.IEvent
import glitch.pubsub.GlitchPubSub

class UnknownResponseEvent(client: GlitchPubSub, val data: JsonElement) : AbstractEvent<GlitchPubSub>(client), IEvent<GlitchPubSub> {
}
